src/compat.rs

```diff
@@ -159,10 +169,66 @@ impl Compatibility {
     }
 }
 
+/// Properly handles runtime unsupported features.
+///
+/// This enables to guarantee consistent behaviors across crate users
+/// and runtime kernels even if this crate get new features.
+/// It eases backward compatibility and enables future-proofness.
+///
+/// Landlock is a security feature designed to help improve security of a running system
+/// thanks to application developers.
+/// To protect users as much as possible,
+/// compatibility with the running system should then be handled in a best-effort way,
+/// contrary to common system features.
+/// In some circumstances
+/// (e.g. applications carefully designed to only be run with a specific set of kernel features),
+/// it may be required to error out if some of these features are not available
+/// and will then not be enforced.
 pub trait Compatible {
-    /// To enable a best-effort security approach, Landlock features that are not supported by the
-    /// running system are silently ignored by default.  If you want to error out when not all your
-    /// requested requirements are met, then you can configure it with `set_best_effort(false)`.
+    /// To enable a best-effort security approach,
+    /// Landlock features that are not supported by the running system
+    /// are silently ignored by default,
+    /// which is a sane choice for most use cases.
+    /// However, on some rare circumstances,
+    /// developers may want to have some guarantees that their applications
+    /// will not run if a certain level of sandboxing is not possible.
+    /// If you really want to error out when not all your requested requirements are met,
+    /// then you can configure it with `set_best_effort(false)`.
+    ///
+    /// The order of this call is important because
+    /// it defines the behavior of the following method calls that return a [`Result`].
+    /// If `set_best_effort(false)` is called on an object,
+    /// then a [`CompatError`] may be returned for the next method calls,
+    /// until the next call to `set_best_effort(true)`.
+    /// This enables to change the behavior of a set of build calls,
+    /// for instance to be sure that the sandbox will at least restrict some access rights.
+    ///
+    /// # Example
+    ///
+    /// Create a ruleset which will at least support execution constraints.
+    ///
+    /// ```
+    /// use landlock::{
+    ///     Access, AccessFs, Compatible, PathBeneath, Ruleset, RulesetCreated, RulesetError, ABI,
+    /// };
+    ///
+    /// fn ruleset_fragile() -> Result<RulesetCreated, RulesetError> {
```

Comment:
This is an enlightening example! I didn't realize that this is how "best effort" mode can be used feature by feature.

(Slight nit: If I understand correctly, two subsequent .handle_access() calls are possible but should not be needed; does it make more sense to use it with .add_rule() calls?)

Reply:
I'm working on changes to `handle_access()` that makes it OR multiple calls, which is convenient for this use case and less error-prone in general.

This example may be better with `add_rule()` but it takes more lines. ;)